exonum-java-binding/qa-service/src/main/java/com/exonum/binding/qaservice/ApiController.java

```diff
@@ -278,6 +289,29 @@ private void getActualConfiguration(RoutingContext rc) {
     respondWithJson(rc, configuration);
   }
 
+  private void getTime(RoutingContext rc) {
+    Optional<ZonedDateTime> time = service.getTime();
+    respondWithJson(rc, time.map(ApiController::convertZdtToString));
+  }
+
+  private void getValidatorsTimes(RoutingContext rc) {
+    Map<PublicKey, ZonedDateTime> validatorsTimes = service.getValidatorsTimes();
+    Map<PublicKey, String> stringValidatorsTimes = convertValidatorsTimesValues(validatorsTimes);
+    respondWithJson(rc, stringValidatorsTimes);
+  }
+
+  @VisibleForTesting
+  static Map<PublicKey, String> convertValidatorsTimesValues(
+      Map<PublicKey, ZonedDateTime> validatorsTimes) {
+    return validatorsTimes.entrySet().stream()
+        .collect(Collectors.toMap(Entry::getKey, t -> convertZdtToString(t.getValue())));
+  }
+
+  @VisibleForTesting
+  static String convertZdtToString(ZonedDateTime zonedDateTime) {
+    return DateTimeFormatter.ofPattern(DATE_TIME_FORMAT).format(zonedDateTime);
```

Comment:
I think it makes sense to extract DateTimeFormatter.ofPattern(DATE_TIME_FORMAT) in a constant, not a DATE_TIME_FORMAT to avoid parsing the pattern repeatedly. It is thread-safe and immutable.